clio-util/src/main/scala/org/broadinstitute/clio/util/auth/AuthUtil.scala

```diff
@@ -18,6 +18,29 @@ object AuthUtil extends ModelAutoDerivation with LazyLogging {
     "https://www.googleapis.com/auth/userinfo.email"
   )
 
+  private val shellOutCreds: Either[Throwable, OAuth2Credentials] = {
+    logger.debug("Shelling out to gcloud to get credentials")
+    Right(ExternalGCloudCredentials)
+  }
+
+  def getCredentials(
+    accessToken: Option[AccessToken],
+    serviceAccountJson: Option[Path]
+  ): Either[Throwable, OAuth2Credentials] = {
+    if (accessToken.isEmpty && serviceAccountJson.isEmpty) {
```

Comment:
Why these up-front `.isEmpty` checks, instead of working the options into the chain of options?